packages/cli/src/ui/contexts/KeypressContext.tsx

```diff
@@ -881,33 +880,29 @@ export function KeypressProvider({
           paste: true,
           sequence: pasteBuffer.toString(),
         });
-        pasteBuffer = Buffer.alloc(0);
+        pasteBuffer = null;
       }
 
-      if (draggingTimerRef.current) {
-        clearTimeout(draggingTimerRef.current);
-        draggingTimerRef.current = null;
-      }
-      if (isDraggingRef.current && dragBufferRef.current) {
+      clearDraggingTimer();
+      if (dragBuffer) {
         broadcast({
           name: '',
           ctrl: false,
           meta: false,
           shift: false,
           paste: true,
-          sequence: dragBufferRef.current,
+          sequence: dragBuffer,
         });
-        isDraggingRef.current = false;
-        dragBufferRef.current = '';
+        dragBuffer = '';
       }
     };
   }, [
     stdin,
```

Comment:
as an aside: having this have dependencies other than stdin is sketchy. We should have the useRef have 0 deps other than perhaps stdin and use useRef to get the latest values if the values happened to change. that can be a follow up. but it is probably more critical now that you are relying on the useEffect only being called exactly once even more to make the business logic work as you rightly deleted the fairly spurious useRefs inside the useEffect.